[建議] errcheck 是否有必要

### 相關 issue
- #90 

### 原本作法的問題 / Existed Problem
以目前 ci 的狀況看起來 errcheck fail 為大宗，目前看起來都是在 call io 相關沒有檢查 bytes 是否不為0 

### 實作細節 / Details of Implement
增加 golangci.yaml 移除 io 相關 errcheck

### 期程 / Schedule
無

### 相關文件 / Documents
https://golangci-lint.run/usage/configuration/

```
internal/delivery/http/route_boards.go:22:11: Error return value of `w.Write` is not checked (errcheck)
w.Write([]byte(`{"error":"token_invalid"}`))
```
以官方文件來說:
https://golang.org/pkg/io/#WriteString

他其實也不會特別處理 io 問題。
我不會特別去驗證官方的 io 相關傳回來的 byte 是否正確，因為我也不知道如何驗證他傳回來的 byte。

> ```
> internal/delivery/http/route_boards.go:22:11: Error return value of `w.Write` is not checked (errcheck)
> w.Write([]byte(`{"error":"token_invalid"}`))
> ```
> 
> 以官方文件來說:
> https://golang.org/pkg/io/#WriteString
> 
> 他其實也不會特別處理 io 問題。
> 我不會特別去驗證官方的 io 相關傳回來的 byte 是否正確，因為我也不知道如何驗證他傳回來的 byte。

測試上為了簡單偶爾會忽略這個檢查 或是確認writer不會回傳err就可以不用檢查
別的官方文件有加註忽略的理由
https://golang.org/pkg/fmt/#Sprintf
production code 都會檢查，以response為例
https://golang.org/src/net/http/response.go#317
![image](https://user-images.githubusercontent.com/7224544/108898468-86a1ad00-7617-11eb-8f2f-84e6d8a48d67.png)


byte多半用做writer內部檢查，正常情況的確不會用到

> ```
> internal/delivery/http/route_boards.go:22:11: Error return value of `w.Write` is not checked (errcheck)
> w.Write([]byte(`{"error":"token_invalid"}`))
> ```
> 
> 以官方文件來說:
> https://golang.org/pkg/io/#WriteString
> 
> 他其實也不會特別處理 io 問題。
> 我不會特別去驗證官方的 io 相關傳回來的 byte 是否正確，因為我也不知道如何驗證他傳回來的 byte。

我認為以 Example 的寫法來作為最佳實作的依據是不可靠的，因為 Example 在寫作時應該會以簡潔扼要為主，對於 Example 的程式碼也不會有測試案例，因此不檢查錯誤也是有可能的。

我認為可以參考的是實際上的 io 套件的原始碼
https://golang.org/src/io/io.go

在 Line:423 的部分確實有提供 Read 以及 Write 在進行錯誤處理的實務上是如何進行的範例
```
		nr, er := src.Read(buf)
		if nr > 0 {
			nw, ew := dst.Write(buf[0:nr])
			if nw < 0 || nr < nw {
				nw = 0
				if ew == nil {
					ew = errInvalidWrite
				}
			}
			written += int64(nw)
			if ew != nil {
				err = ew
				break
			}
			if nr != nw {
				err = ErrShortWrite
				break
			}
		}
		if er != nil {
			if er != EOF {
				err = er
			}
			break
		}
		
```

> > ```
> > internal/delivery/http/route_boards.go:22:11: Error return value of `w.Write` is not checked (errcheck)
> > w.Write([]byte(`{"error":"token_invalid"}`))
> > ```
> > 
> > 
> > 以官方文件來說:
> > https://golang.org/pkg/io/#WriteString
> > 他其實也不會特別處理 io 問題。
> > 我不會特別去驗證官方的 io 相關傳回來的 byte 是否正確，因為我也不知道如何驗證他傳回來的 byte。
> 
> 我認為以 Example 的寫法來作為最佳實作的依據是不可靠的，因為 Example 在寫作時應該會以簡潔扼要為主，對於 Example 的程式碼也不會有測試案例，因此不檢查錯誤也是有可能的。
> 
> 我認為可以參考的是實際上的 io 套件的原始碼
> https://golang.org/src/io/io.go
> 
> 在 Line:423 的部分確實有提供 Read 以及 Write 在進行錯誤處理的實務上是如何進行的範例
> 
> ```
> 		nr, er := src.Read(buf)
> 		if nr > 0 {
> 			nw, ew := dst.Write(buf[0:nr])
> 			if nw < 0 || nr < nw {
> 				nw = 0
> 				if ew == nil {
> 					ew = errInvalidWrite
> 				}
> 			}
> 			written += int64(nw)
> 			if ew != nil {
> 				err = ew
> 				break
> 			}
> 			if nr != nw {
> 				err = ErrShortWrite
> 				break
> 			}
> 		}
> 		if er != nil {
> 			if er != EOF {
> 				err = er
> 			}
> 			break
> 		}
> 		
> ```

補充一點，一般使用不需要檢查byte長度，pichu這個範例是因為需要利用byte長度比對判斷後回傳特定的error
在Writer的interface上有說明了，傳入跟傳出的長度不同會拋出錯誤

![image](https://user-images.githubusercontent.com/7224544/108962823-97830a80-7679-11eb-9ced-56899386a9ec.png)

--- a/internal/delivery/http/route_boards.go
+++ b/internal/delivery/http/route_boards.go
@@ -21,7 +21,7 @@ func (delivery *httpDelivery) getBoardList(w http.ResponseWriter, r *http.Reques
 			w.WriteHeader(http.StatusUnauthorized)
 			_, err := w.Write([]byte(`{"error":"token_invalid"}`))
 			if err != nil {
-				delivery.logger.Errorf("getBoardList write token invalid response err : %w", err)
+				delivery.logger.Errorf("getBoardList write token invalid response err: %w", err)
 			}
 			return
 		} else {
@@ -43,7 +43,7 @@ func (delivery *httpDelivery) getBoardList(w http.ResponseWriter, r *http.Reques
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getBoardList write response err : %w", err)
+		delivery.logger.Errorf("getBoardList write response err: %w", err)
 	}
 }
 
@@ -62,7 +62,7 @@ func (delivery *httpDelivery) getPopularBoardList(w http.ResponseWriter, r *http
 		w.WriteHeader(http.StatusInternalServerError)
 		_, err := w.Write(b)
 		if err != nil {
-			delivery.logger.Errorf("getPopularBoardList write error response err : %w", err)
+			delivery.logger.Errorf("getPopularBoardList write error response err: %w", err)
 		}
 		return
 	}
@@ -81,7 +81,7 @@ func (delivery *httpDelivery) getPopularBoardList(w http.ResponseWriter, r *http
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getPopularBoardList write success response err : %w", err)
+		delivery.logger.Errorf("getPopularBoardList write success response err: %w", err)
 	}
 }
 
@@ -112,7 +112,7 @@ func (delivery *httpDelivery) getBoardInformation(w http.ResponseWriter, r *http
 		b, _ := json.MarshalIndent(m, "", "  ")
 		_, err = w.Write(b)
 		if err != nil {
-			delivery.logger.Errorf("getBoardInformation write error response err : %w", err)
+			delivery.logger.Errorf("getBoardInformation write error response err: %w", err)
 		}
 		return
 	}
@@ -124,7 +124,7 @@ func (delivery *httpDelivery) getBoardInformation(w http.ResponseWriter, r *http
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getBoardInformation write success response err : %w", err)
+		delivery.logger.Errorf("getBoardInformation write success response err: %w", err)
 	}
 }
 

--- a/internal/delivery/http/route_boards.go
+++ b/internal/delivery/http/route_boards.go
@@ -19,7 +19,10 @@ func (delivery *httpDelivery) getBoardList(w http.ResponseWriter, r *http.Reques
 		// Support Guest?
 		if !supportGuest() {
 			w.WriteHeader(http.StatusUnauthorized)
-			w.Write([]byte(`{"error":"token_invalid"}`))
+			_, err := w.Write([]byte(`{"error":"token_invalid"}`))
+			if err != nil {
+				delivery.logger.Errorf("getBoardList write token invalid response err : %w", err)
+			}
 			return
 		} else {
 			userId = "guest" // TODO: use const variable
@@ -38,7 +41,10 @@ func (delivery *httpDelivery) getBoardList(w http.ResponseWriter, r *http.Reques
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getBoardList write response err : %w", err)
+	}
 }
 
 func (delivery *httpDelivery) getPopularBoardList(w http.ResponseWriter, r *http.Request) {
@@ -54,7 +60,10 @@ func (delivery *httpDelivery) getPopularBoardList(w http.ResponseWriter, r *http
 		}
 		b, _ := json.MarshalIndent(m, "", "  ")
 		w.WriteHeader(http.StatusInternalServerError)
-		w.Write(b)
+		_, err := w.Write(b)
+		if err != nil {
+			delivery.logger.Errorf("getPopularBoardList write error response err : %w", err)
+		}
 		return
 	}
 
@@ -66,11 +75,14 @@ func (delivery *httpDelivery) getPopularBoardList(w http.ResponseWriter, r *http
 	responseMap := map[string]interface{}{
 		"data": struct {
 			Items []interface{} `json:"items"`
-		}{ dataList },
+		}{dataList},
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getPopularBoardList write success response err : %w", err)
+	}
 }
 
 func (delivery *httpDelivery) getBoardInformation(w http.ResponseWriter, r *http.Request, boardId string) {
@@ -98,7 +110,10 @@ func (delivery *httpDelivery) getBoardInformation(w http.ResponseWriter, r *http
 			"error_description": "get board for " + boardId + " failed",
 		}
 		b, _ := json.MarshalIndent(m, "", "  ")
-		w.Write(b)
+		_, err = w.Write(b)
+		if err != nil {
+			delivery.logger.Errorf("getBoardInformation write error response err : %w", err)
+		}
 		return
 	}
 
@@ -107,7 +122,10 @@ func (delivery *httpDelivery) getBoardInformation(w http.ResponseWriter, r *http
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getBoardInformation write success response err : %w", err)
+	}
 }
 
 // marshal generate board or class metadata object,

--- a/internal/delivery/http/route_boards_articles.go
+++ b/internal/delivery/http/route_boards_articles.go
@@ -68,7 +68,7 @@ func (delivery *httpDelivery) getBoardArticles(w http.ResponseWriter, r *http.Re
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getBoardArticles write success response err : %w", err)
+		delivery.logger.Errorf("getBoardArticles write success response err: %w", err)
 	}
 }
 
@@ -104,6 +104,6 @@ func (delivery *httpDelivery) getBoardArticlesFile(w http.ResponseWriter, r *htt
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getBoardArticlesFile write success response err : %w", err)
+		delivery.logger.Errorf("getBoardArticlesFile write success response err: %w", err)
 	}
 }

--- a/internal/delivery/http/route_boards_articles.go
+++ b/internal/delivery/http/route_boards_articles.go
@@ -66,8 +66,10 @@ func (delivery *httpDelivery) getBoardArticles(w http.ResponseWriter, r *http.Re
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
-
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getBoardArticles write success response err : %w", err)
+	}
 }
 
 func (delivery *httpDelivery) getBoardArticlesFile(w http.ResponseWriter, r *http.Request, boardId string, filename string) {
@@ -100,5 +102,8 @@ func (delivery *httpDelivery) getBoardArticlesFile(w http.ResponseWriter, r *htt
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getBoardArticlesFile write success response err : %w", err)
+	}
 }

--- a/internal/delivery/http/route_boards_test.go
+++ b/internal/delivery/http/route_boards_test.go
@@ -28,7 +28,10 @@ func TestGetPopularBoardList(t *testing.T) {
 	}
 
 	responseMap := map[string]interface{}{}
-	json.Unmarshal(rr.Body.Bytes(), &responseMap)
+	err = json.Unmarshal(rr.Body.Bytes(), &responseMap)
+	if err != nil {
+		t.Errorf("get unexpect json : %w", err)
+	}
 	t.Logf("got response %v", rr.Body.String())
 	responseData := responseMap["data"]
 	popularBoards := responseData.(map[string]interface{})["items"].([]interface{})

--- a/internal/delivery/http/route_boards_test.go
+++ b/internal/delivery/http/route_boards_test.go
@@ -30,7 +30,7 @@ func TestGetPopularBoardList(t *testing.T) {
 	responseMap := map[string]interface{}{}
 	err = json.Unmarshal(rr.Body.Bytes(), &responseMap)
 	if err != nil {
-		t.Errorf("get unexpect json : %w", err)
+		t.Errorf("get unexpect json: %w", err)
 	}
 	t.Logf("got response %v", rr.Body.String())
 	responseData := responseMap["data"]

--- a/internal/delivery/http/route_boards_treasures.go
+++ b/internal/delivery/http/route_boards_treasures.go
@@ -47,8 +47,10 @@ func (delivery *httpDelivery) getBoardTreasures(w http.ResponseWriter, r *http.R
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
-
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getBoardTreasures write success response err : %w", err)
+	}
 }
 
 func (delivery *httpDelivery) getBoardTreasuresFile(w http.ResponseWriter, r *http.Request, boardId string, treasuresId []string, filename string) {

--- a/internal/delivery/http/route_boards_treasures.go
+++ b/internal/delivery/http/route_boards_treasures.go
@@ -49,7 +49,7 @@ func (delivery *httpDelivery) getBoardTreasures(w http.ResponseWriter, r *http.R
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getBoardTreasures write success response err : %w", err)
+		delivery.logger.Errorf("getBoardTreasures write success response err: %w", err)
 	}
 }
 

--- a/internal/delivery/http/route_classes.go
+++ b/internal/delivery/http/route_classes.go
@@ -45,7 +45,7 @@ func (delivery *httpDelivery) getClassesList(w http.ResponseWriter, r *http.Requ
 			w.WriteHeader(http.StatusUnauthorized)
 			_, err = w.Write([]byte(`{"error":"token_invalid"}`))
 			if err != nil {
-				delivery.logger.Errorf("getClassesList write error response err : %w", err)
+				delivery.logger.Errorf("getClassesList write error response err: %w", err)
 			}
 			return
 		} else {
@@ -71,6 +71,6 @@ func (delivery *httpDelivery) getClassesList(w http.ResponseWriter, r *http.Requ
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("getClassesList write success response err : %w", err)
+		delivery.logger.Errorf("getClassesList write success response err: %w", err)
 	}
 }

--- a/internal/delivery/http/route_classes.go
+++ b/internal/delivery/http/route_classes.go
@@ -43,7 +43,10 @@ func (delivery *httpDelivery) getClassesList(w http.ResponseWriter, r *http.Requ
 		// Support Guest?
 		if !supportGuest() {
 			w.WriteHeader(http.StatusUnauthorized)
-			w.Write([]byte(`{"error":"token_invalid"}`))
+			_, err = w.Write([]byte(`{"error":"token_invalid"}`))
+			if err != nil {
+				delivery.logger.Errorf("getClassesList write error response err : %w", err)
+			}
 			return
 		} else {
 			userId = "guest" // TODO: use const variable
@@ -66,5 +69,8 @@ func (delivery *httpDelivery) getClassesList(w http.ResponseWriter, r *http.Requ
 	}
 
 	b, _ := json.MarshalIndent(responseMap, "", "  ")
-	w.Write(b)
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("getClassesList write success response err : %w", err)
+	}
 }

--- a/internal/delivery/http/route_classes_test.go
+++ b/internal/delivery/http/route_classes_test.go
@@ -33,7 +33,6 @@ func TestGetClassesList(t *testing.T) {
 
 	responseMap := map[string][]interface{}{}
 	err = json.Unmarshal(rr.Body.Bytes(), &responseMap)
-
 	if err != nil {
 		t.Errorf("get unexpect json : %w", err)
 	}

--- a/internal/delivery/http/route_classes_test.go
+++ b/internal/delivery/http/route_classes_test.go
@@ -35,7 +35,7 @@ func TestGetClassesList(t *testing.T) {
 	err = json.Unmarshal(rr.Body.Bytes(), &responseMap)
 
 	if err != nil {
-		t.Errorf("get unexpect json : %w", err)
+		t.Errorf("get unexpect json: %w", err)
 	}
 
 	t.Logf("got response %v", rr.Body.String())

--- a/internal/delivery/http/route_classes_test.go
+++ b/internal/delivery/http/route_classes_test.go
@@ -32,7 +32,12 @@ func TestGetClassesList(t *testing.T) {
 	}
 
 	responseMap := map[string][]interface{}{}
-	json.Unmarshal(rr.Body.Bytes(), &responseMap)
+	err = json.Unmarshal(rr.Body.Bytes(), &responseMap)
+
+	if err != nil {
+		t.Errorf("get unexpect json : %w", err)
+	}
+
 	t.Logf("got response %v", rr.Body.String())
 	responseData := responseMap["data"]
 	for i, d := range responseData {

--- a/internal/delivery/http/route_token.go
+++ b/internal/delivery/http/route_token.go
@@ -15,7 +15,7 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 	err := r.ParseForm()
 
 	if err == nil {
-		delivery.logger.Errorf("postToken parse form err : %w", err)
+		delivery.logger.Errorf("postToken parse form err: %w", err)
 	}
 
 	username := r.FormValue("username")
@@ -30,7 +30,7 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 		b, _ := json.MarshalIndent(m, "", "  ")
 		_, err = w.Write(b)
 		if err != nil {
-			delivery.logger.Errorf("postToken write get user id error response err : %w", err)
+			delivery.logger.Errorf("postToken write get user id error response err: %w", err)
 		}
 		return
 
@@ -49,7 +49,7 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 		w.WriteHeader(http.StatusUnauthorized)
 		_, err = w.Write(b)
 		if err != nil {
-			delivery.logger.Errorf("postToken write password verify error response err : %w", err)
+			delivery.logger.Errorf("postToken write password verify error response err: %w", err)
 		}
 		return
 	}
@@ -65,7 +65,7 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 
 	_, err = w.Write(b)
 	if err != nil {
-		delivery.logger.Errorf("postToken success response err : %w", err)
+		delivery.logger.Errorf("postToken success response err: %w", err)
 	}
 }
 

--- a/internal/delivery/http/route_token.go
+++ b/internal/delivery/http/route_token.go
@@ -13,9 +13,8 @@ import (
 func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) {
 
 	err := r.ParseForm()
-
 	if err == nil {
-		delivery.logger.Errorf("postToken parse form err : %w", err)
+		delivery.logger.Errorf("postToken parse form err: %w", err)
 	}
 
 	username := r.FormValue("username")

--- a/internal/delivery/http/route_token.go
+++ b/internal/delivery/http/route_token.go
@@ -12,7 +12,11 @@ import (
 
 func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) {
 
-	r.ParseForm()
+	err := r.ParseForm()
+
+	if err == nil {
+		delivery.logger.Errorf("postToken parse form err : %w", err)
+	}
 
 	username := r.FormValue("username")
 	password := r.FormValue("password")
@@ -24,7 +28,10 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 			"error_description": err.Error(),
 		}
 		b, _ := json.MarshalIndent(m, "", "  ")
-		w.Write(b)
+		_, err = w.Write(b)
+		if err != nil {
+			delivery.logger.Errorf("postToken write get user id error response err : %w", err)
+		}
 		return
 
 	}
@@ -40,7 +47,10 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 		}
 		b, _ := json.MarshalIndent(m, "", "  ")
 		w.WriteHeader(http.StatusUnauthorized)
-		w.Write(b)
+		_, err = w.Write(b)
+		if err != nil {
+			delivery.logger.Errorf("postToken write password verify error response err : %w", err)
+		}
 		return
 	}
 
@@ -53,7 +63,10 @@ func (delivery *httpDelivery) postToken(w http.ResponseWriter, r *http.Request) 
 
 	b, _ := json.MarshalIndent(m, "", "  ")
 
-	w.Write(b)
+	_, err = w.Write(b)
+	if err != nil {
+		delivery.logger.Errorf("postToken success response err : %w", err)
+	}
 }
 
 func (delivery *httpDelivery) verifyPassword(userec bbs.UserRecord, password string) error {

--- a/internal/delivery/http/route_users.go
+++ b/internal/delivery/http/route_users.go
@@ -47,7 +47,7 @@ func (delivery *httpDelivery) getUserInformation(w http.ResponseWriter, r *http.
 		b, _ := json.MarshalIndent(m, "", "  ")
 		_, err = w.Write(b)
 		if err != nil {
-			delivery.logger.Errorf("getUserInformation error response err : %w", err)
+			delivery.logger.Errorf("getUserInformation error response err: %w", err)
 		}
 		return
 	}
@@ -59,7 +59,7 @@ func (delivery *httpDelivery) getUserInformation(w http.ResponseWriter, r *http.
 
 	_, err = w.Write(responseByte)
 	if err != nil {
-		delivery.logger.Errorf("getUserInformation success response err : %w", err)
+		delivery.logger.Errorf("getUserInformation success response err: %w", err)
 	}
 }
 
@@ -92,6 +92,6 @@ func (delivery *httpDelivery) getUserFavorites(w http.ResponseWriter, r *http.Re
 
 	_, err = w.Write(responseByte)
 	if err != nil {
-		delivery.logger.Errorf("getUserFavorites success response err : %w", err)
+		delivery.logger.Errorf("getUserFavorites success response err: %w", err)
 	}
 }

--- a/internal/delivery/http/route_users.go
+++ b/internal/delivery/http/route_users.go
@@ -45,7 +45,10 @@ func (delivery *httpDelivery) getUserInformation(w http.ResponseWriter, r *http.
 			"error_description": err.Error(),
 		}
 		b, _ := json.MarshalIndent(m, "", "  ")
-		w.Write(b)
+		_, err = w.Write(b)
+		if err != nil {
+			delivery.logger.Errorf("getUserInformation error response err : %w", err)
+		}
 		return
 	}
 
@@ -54,7 +57,10 @@ func (delivery *httpDelivery) getUserInformation(w http.ResponseWriter, r *http.
 	}
 	responseByte, _ := json.MarshalIndent(responseMap, "", "  ")
 
-	w.Write(responseByte)
+	_, err = w.Write(responseByte)
+	if err != nil {
+		delivery.logger.Errorf("getUserInformation success response err : %w", err)
+	}
 }
 
 func (delivery *httpDelivery) getUserFavorites(w http.ResponseWriter, r *http.Request, userId string) {
@@ -84,5 +90,8 @@ func (delivery *httpDelivery) getUserFavorites(w http.ResponseWriter, r *http.Re
 
 	responseByte, _ := json.MarshalIndent(responseMap, "", "  ")
 
-	w.Write(responseByte)
+	_, err = w.Write(responseByte)
+	if err != nil {
+		delivery.logger.Errorf("getUserFavorites success response err : %w", err)
+	}
 }

--- a/internal/delivery/http/route_users_test.go
+++ b/internal/delivery/http/route_users_test.go
@@ -32,7 +32,12 @@ func TestGetUserInformation(t *testing.T) {
 	}
 
 	responsedMap := map[string]interface{}{}
-	json.Unmarshal(rr.Body.Bytes(), &responsedMap)
+	err = json.Unmarshal(rr.Body.Bytes(), &responsedMap)
+
+	if err != nil {
+		t.Errorf("get unexpect json : %w", err)
+	}
+
 	t.Logf("got response %v", rr.Body.String())
 	responsedData := responsedMap["data"].(map[string]interface{})
 	if responsedData["user_id"] != userID {

--- a/internal/delivery/http/route_users_test.go
+++ b/internal/delivery/http/route_users_test.go
@@ -35,7 +35,7 @@ func TestGetUserInformation(t *testing.T) {
 	err = json.Unmarshal(rr.Body.Bytes(), &responsedMap)
 
 	if err != nil {
-		t.Errorf("get unexpect json : %w", err)
+		t.Errorf("get unexpect json: %w", err)
 	}
 
 	t.Logf("got response %v", rr.Body.String())

--- a/internal/delivery/http/route_users_test.go
+++ b/internal/delivery/http/route_users_test.go
@@ -33,7 +33,6 @@ func TestGetUserInformation(t *testing.T) {
 
 	responsedMap := map[string]interface{}{}
 	err = json.Unmarshal(rr.Body.Bytes(), &responsedMap)
-
 	if err != nil {
 		t.Errorf("get unexpect json : %w", err)
 	}

